src/main/daml/Daml/Finance/Interface/Claims/Dynamic/Instrument.daml

```diff
@@ -42,12 +42,15 @@ interface Instrument requires BaseInstrument.I, Disclosure.I where
       pure $ view this
 
   nonconsuming choice CreateNewVersion : ContractId Instrument
-    -- ^ Create a new version of an instrument, using a new lastEventTimestamp.
+    -- ^ Create a new version of an instrument, using a new lastEventTimestamp and a list of
+    -- previous elections (if applicable).
     with
       version : Text
         -- ^ The new version of the instrument.
       lastEventTimestamp : Time
         -- ^ The new lastEventTimestamp of the instrument.
+      prevElections : [EventData]
```

Comment:
Given that we are touching the interface, can we rename this to `previousEvents`?

It applies to other type of instruments, such as a barrier options, where we need to store the `barrierHit` time and replay it when lifecycling.

An alternative is to replace both `lastEventTimestamp` and `prevElections` with `latestEvent`. The implementation of `CreateNewVersion` is then responsible to do the following:
- in the case of a bond, just store the `lastEventTimestamp`, we don't care about anything elsde
- in the case of a callable bond, store the `lastEventTimestamp` and the election history
- in the case of a barrier option, store all events to be replayed